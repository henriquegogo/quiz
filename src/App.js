import React, { Component, Fragment } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import Header from './components/Header';
import Categories from './pages/Categories';
import Trivia from './pages/Trivia';
import Report from './pages/Report';
import './App.scss';

class App extends Component {
  render() {
    return (
      <Fragment>
        <Header />
        <main>
          <Router>
            <Route path='/' exact render={() => <Categories />} />
            <Route path='/trivia' render={() => <Trivia />} />
            <Route path='/report' render={() => <Report />} />
          </Router>
        </main>
      </Fragment>
    );
  }
}

export default App;
