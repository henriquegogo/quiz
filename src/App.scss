$medium-screen: 768px;
$small-screen: 576px;
$smaller-screen: 375px;

main {
  margin: 0 auto;
  padding: 10px;
  max-width: 992px;
  color: #1E2124;

  h1 {
    margin: 30px 0 20px;
    font-size: 44px;
    font-weight: normal;
    line-height: 56px;
  }

  h2 {
    margin: 20px 0;
    font-size: 24px;
    font-weight: normal;
    line-height: 32px;
  }
}

section {
  padding: 40px;
  font-weight: 300;
  background: #FFFFFF;
  border: 1px solid #E4E4E6;
  box-sizing: border-box;
  box-shadow: 0px 1px 4px rgba(52, 60, 88, 0.4);
  border-radius: 8px;

  h3 {
    margin: 0 0 10px;
    font-size: 18px;
    font-weight: normal;
    line-height: 24px;
  }

  p {
    font-size: 16px;
    line-height: 22px;
    letter-spacing: 0.2px;
  }
}

.answers {
  margin: 30px 0 0;
  width: 100%;

  label {
    span {
      display: block;
      margin: 16px 0;
      padding: 16px;
      cursor: pointer;
      background: #FFFFFF;
      min-height: 60px;
      border: 1px solid #C9CCCF;
      box-sizing: border-box;
      box-shadow: 0px 1px 2px rgba(52, 60, 88, 0.4);
      border-radius: 8px;
    }
  }

  input[type='radio'] {
    display: none;

    &:checked + span {
      border: 3px solid #0467DB;
      box-shadow: 0px 1px 2px rgba(52, 60, 88, 0.4);
    }
  }

  button[type='submit'] {
    display: block;
    margin: 40px auto 0;
    padding: 12px 24px;
    cursor: pointer;
    color: #FFFFFF;
    font-size: 16px;
    line-height: 24px;
    border: none;
    border-radius: 8px;
    background: #0467DB;
    box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.15);

    &[disabled] {
      cursor: default;
      background: #C9CCCF;
    }

    @media (max-width: $medium-screen) {
      width: 100%;
    }
  }
}

.card-group {
  margin: -13px;
  display: flex;
  flex-wrap: wrap;

  .card {
    display: inline-flex;
    align-items: flex-end;
    margin: 13px;
    padding: 16px;
    width: calc(25% -  60px);
    height: 70px;
    vertical-align: top;
    font-size: 14px;
    text-decoration: none;
    color: #1E2124;
    background: #ffffff;
    border: 1px solid #E4E4E6;
    box-shadow: 0px 1px 4px rgba(52, 60, 88, 0.4);
    border-radius: 8px;
  }

  @media (max-width: $small-screen) {
    margin: -8px;

    .card {
      margin: 8px;
      width: calc(33% -  50px);
    }
  }

  @media (max-width: $smaller-screen) {
    .card {
      width: calc(50% -  50px);
    }
  }
}

