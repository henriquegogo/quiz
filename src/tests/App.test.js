import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

test('App renderize properly', () => {
	const component = shallow(<App />);

	expect(component).toMatchSnapshot();
});
