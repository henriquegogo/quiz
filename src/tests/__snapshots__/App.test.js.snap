// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renderize properly 1`] = `ShallowWrapper {}`;
